crates/syn-solidity/src/yul/expr/fn_call.rs

```diff
@@ -0,0 +1,105 @@
+use crate::{Spanned, YulEVMBuiltIn, YulExpr, YulIdent};
+
+use proc_macro2::Span;
+use std::fmt;
+use syn::{
+    parenthesized,
+    parse::{discouraged::Speculative, Parse, ParseStream, Result},
+    punctuated::Punctuated,
+    token::Paren,
+    Token,
+};
+
+/// Yul function call.
+///
+/// Solidity Reference:
+/// <https://docs.soliditylang.org/en/latest/grammar.html#a4.SolidityParser.yulFunctionCall>
+#[derive(Clone)]
+pub struct YulFnCall {
+    pub function_type: FnType,
+    pub paren_token: Paren,
+    pub arguments: Punctuated<YulExpr, Token![,]>,
+}
+
+impl Parse for YulFnCall {
+    fn parse(input: ParseStream<'_>) -> Result<Self> {
+        let content;
+        Ok(Self {
+            function_type: input.parse()?,
+            paren_token: parenthesized!(content in input),
+            arguments: content.parse_terminated(YulExpr::parse, Token![,])?,
+        })
+    }
+}
+
+impl Spanned for YulFnCall {
+    fn span(&self) -> Span {
+        let span = self.function_type.span();
+        span.join(self.arguments.span()).unwrap_or(span)
+    }
+
+    fn set_span(&mut self, span: Span) {
+        self.function_type.set_span(span);
+        self.paren_token = Paren(span);
+        self.arguments.set_span(span);
+    }
+}
+
+impl fmt::Debug for YulFnCall {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        f.debug_struct("YulFnCall")
+            .field("function_type", &self.function_type)
+            .field("paren_token", &self.paren_token)
+            .field("arguments", &self.arguments)
+            .finish()
+    }
+}
```

Comment:
The purpose of manually implementing `Debug` was to skip the token fields since they don't add anything useful to the output